test/common/network/udp_listener_impl_test_base.h

```diff
@@ -64,12 +64,7 @@ class UdpListenerImplTestBase : public ListenerImplTestBase {
     if (version_ == Address::IpVersion::v4) {
       // Linux kernel regards any 127.x.x.x as local address. But Mac OS doesn't.
       send_from_addr = std::make_shared<Address::Ipv4Instance>(
-#ifndef __APPLE__
-          "127.1.2.3",
-#else
-          "127.0.0.1",
-#endif
-          server_socket_->localAddress()->ip()->port());
+          "127.0.0.1", server_socket_->localAddress()->ip()->port());
```

Comment:
It appears this was deliberate to avoid confusion between pings to/from .0.0.1 (self) and an alternate/'external' IP address for purpose of testing. Did we want to preserve this test by abstracting an alternate localhost address? Or was this all conflated into the IP_FREEBIND constraints mentioned below for IPv6 causing windows test to fail?

Reply:
We use this address to compare with the one that we receive data from. I thought this change was necessary since we changed from anyAddress to LoopbackAddress.

Am I missing something?